src/interface/src/app/treatments/treatments.state.ts

```diff
@@ -119,6 +119,15 @@ export class TreatmentsState {
       );
   }
 
+  // simpler, no side-effect summary loader...
+  reloadSummary() {
+    this.treatmentsService
```

Comment:
Here instead of subscribing directly, you should pipe and map.
That way we can return the promise here so we can pipe it on the component (for errors and whatnot).